### PR TITLE
Compose workspace

### PR DESCRIPTION
**Pull request by LokiFrostGiant**
For review by Rosário Fernandes @thatfiredev .

Created an option to run Admob using Jetpack Compose. Separated functionalities for interstitial ads to increase flexibility, such as setting callback, initializing, and displaying. Demonstrates each functionality more clearly/isolated. Interstitial is now re-callable by this.

In the MainComposeActivity file that was created, please review Composables to advise on whether to continue passing Lambdas or if passing a Context would be better suitable in this scenario. Additionally, please review the Theme.kt and Color.kt files to suggest any changes that properly meet your expected standards/format.

Please note that any changes to Config have been undone for a separate PR by @millandavid .

### CODE REVIEW

**Review by thatfiredev — COMMENTED** [review]
Thanks for starting this, @LokiFrostGiant!
I've added small comments before I review the Kotlin code.

File: admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainComposeActivity.kt [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        setContent {
+
+            AdmobTheme {
+                 //A surface container using the 'background' color from the theme
+                Surface(
+                    modifier = Modifier.fillMaxSize(),
+                    color = MaterialTheme.colorScheme.background
+                ) {
+//                    Greeting("Android")
+                    MainAppView( buttonClickEventAdLoader = buttonClickLambda ) // better here?
```
Comment (thatfiredev) [inline admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainComposeActivity.kt:47]: This is better :)

And please remove the comments when you make the change.

File: admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainComposeActivity.kt [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    Text(text = "Hello $name!")
+}
+
+@Preview(showBackground = true)
+@Composable
+fun DefaultPreview() {
+    AdmobTheme {
+        Greeting("Android")
+    }
+}
```
Comment (thatfiredev) [inline admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainComposeActivity.kt:208]: Seems like we're no longer using these. We can delete them :)

File: .gitignore [hunk 3]
```diff
@@ -13,3 +13,4 @@ google-services.json
 .settings
 .classpath
 .vscode
+config/local-Millans-AsusG14Zephyrus.properties
```
[the remark this answers is not in the document]
Comment (LokiFrostGiant) [reply .gitignore:16]: this appeared to be a residual from an automatic action on Millan's machine. It has now been removed.

File: admob/app/build.gradle [hunk 4]
```diff
@@ -71,6 +71,7 @@ dependencies {
     // For an optimal experience using AdMob, add the Firebase SDK
     // for Google Analytics. This is recommended, but not required.
     implementation 'com.google.firebase:firebase-analytics'
+    implementation 'androidx.compose.material3:material3:1.0.0-alpha02'
```
[the remark this answers is not in the document]
Comment (LokiFrostGiant) [reply admob/app/build.gradle:74]: Version corrected and build successful.

File: admob/build.gradle [hunk 5]
```diff
@@ -1,6 +1,9 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
 buildscript {
+    ext {
+        compose_version = '1.1.0-beta01'
```
[the remark this answers is not in the document]
Comment (LokiFrostGiant) [reply admob/build.gradle:5]: Version corrected and build successful.

File: admob/app/src/main/java/com/google/samples/quickstart/admobexample/EntryChoiceActivity.kt [hunk 6]
```diff
@@ -12,11 +12,15 @@ class EntryChoiceActivity : BaseEntryChoiceActivity() {
...
-                        Intent(this, MainActivity::class.java)),
+                        Intent(this, com.google.samples.quickstart.admobexample.java.MainActivity::class.java)),
                 Choice(
                         "Kotlin",
                         "Run the Firebase Admob quickstart written in Kotlin.",
-                        Intent(this, com.google.samples.quickstart.admobexample.kotlin.MainActivity::class.java))
+                        Intent(this, com.google.samples.quickstart.admobexample.kotlin.MainActivity::class.java)),
+                Choice(
+                        "Compose",
+                        "Run the Firebase Admob quickstart written in Kotlin with Compose.",
```
[the remark this answers is not in the document]
Comment (LokiFrostGiant) [reply admob/app/src/main/java/com/google/samples/quickstart/admobexample/EntryChoiceActivity.kt:22]: Entry choice description updated.

File: admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainComposeActivity.kt (hunk 1 above)
Comment (LokiFrostGiant) [reply admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainComposeActivity.kt:47]: Cleaned up!

File: admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainComposeActivity.kt (hunk 2 above)
Comment (LokiFrostGiant) [reply admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainComposeActivity.kt:208]: Default Composables removed.

File: admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainComposeActivity.kt [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+                  },
+        bottomBar = { // keeps the banner ad at the bottom!
+            AdvertBanner()
+        }
+    )
+
+}
+
+@Composable
+fun InterstitialButton(modifier: Modifier = Modifier, myClickEventInterstitialLoader : () -> Unit = {}){
```
Comment (thatfiredev) [inline admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainComposeActivity.kt:180]: I think we can declare this Button on the MainAppView, no need to separate it.

**Review by thatfiredev — COMMENTED** [review]
@LokiFrostGiant  our [CI](https://github.com/firebase/quickstart-android/actions/runs/4106979296/jobs/7129921066) seems to be mad about some indentation errors, can you please take a look?

I have also added some comments bellow.

File: admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainComposeActivity.kt (hunk 7 above)
Comment (LokiFrostGiant) [reply admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainComposeActivity.kt:180]: As long as it doesn't look too big or too long for you by having it all together, I shall gladly do so.

File: admob/app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -12,5 +12,6 @@
     <string name="admob_app_id">ca-app-pub-3940256099942544~3347511713</string>
 
     <string name="banner_ad_unit_id">ca-app-pub-3940256099942544/6300978111</string>
-    <string name="interstitial_ad_unit_id">ca-app-pub-3940256099942544/1033173712</string>
+    <string name="interstitial_ad_unit_id">ca-app-pub-3940256099942544/1033173712</string>\
+    <string name="title_activity_main_compose">MainComposeActivity</string>
```
[the remark this answers is not in the document]
Comment (LokiFrostGiant) [reply admob/app/src/main/res/values/strings.xml:16]: Hi, all these strings are in fact being used. interstitial_ad_unit_id is used in [MainComposeActivity](https://github.com/LokiFrostGiant/quickstart-android/blob/947700daa9a05fa29fa7a6fc6807c1397a162509/admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainComposeActivity.kt#L54)

**Comment by LokiFrostGiant** [comment]
Hi @thatfiredev , I made the changes, but I expect the indentation errors will happen again. Once the CI runs again and reproduces the errors for the new version of the app, please let me know and I will revisit it again. Thanks!

File: admob/app/src/main/res/values/strings.xml (hunk 8 above)
[the remark this answers is not in the document]
Comment (thatfiredev) [reply admob/app/src/main/res/values/strings.xml:16]: Sorry, I meant `title_activity_main_compose`.

**Review by thatfiredev — APPROVED** [review]
Thanks @LokiFrostGiant !

**Pull request merged**